lib/wasi/src/syscalls/mod.rs

```diff
@@ -2659,18 +2659,12 @@ pub fn path_rename<M: MemorySize>(
         source_path.to_str().as_ref().unwrap(),
         true
     ));
-    if state
-        .fs
-        .get_inode_at_path(
-            inodes.deref_mut(),
-            new_fd,
-            target_path.to_str().as_ref().unwrap(),
-            true,
-        )
-        .is_ok()
-    {
-        return Errno::Exist;
-    }
+    state.fs.get_inode_at_path(
+        inodes.deref_mut(),
+        new_fd,
+        target_path.to_str().as_ref().unwrap(),
```

Comment:
What happens when the target path isn't a UTF-8 string?

Reply:
The rest of the code uses `unwrap()` when handling paths, so to *properly* handle paths that aren't UTF-8 we'd need to rewrite a lot of code.